docs/external-plugin-messaging.md

```diff
@@ -0,0 +1,87 @@
+
+The `Map<String, Object>` that is supplied to `PluginMessage` will be converted into [`ExternalNotificationRequest`](../src/main/java/dinkplugin/domain/ExternalNotificationRequest.java).
+
+| Field            | Required | Type             | Description                                                                                                                                                                             |
+| ---------------- | -------- | ---------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `text`           | Y        | String           | The body text of the notification. This field supports templating (see `replacements` below) and by default `%USERNAME%` is an available replacement.                                   |
+| `url`            | N        | String           | The Discord URLs that the notification should be sent to (newline separated).                                                                                                           |
+| `title`          | N        | String           | The title for the Discord embed.                                                                                                                                                        |
+| `thumbnail`      | N        | String           | A URL to an image for the thumbnail icon of the Discord embed.                                                                                                                          |
+| `footer`         | N        | String           | The footer text for the Discord embed.                                                                                                                                                  |
+| `imageRequested` | N        | boolean          | Whether dink should include a screenshot with the notification.                                                                                                                         |
+| `image`          | N        | `java.awt.Image` | The image to use in place of a screenshot.                                                                                                                                              |
+| `fields`         | N        | List             | A list of [embed fields](https://discord.com/developers/docs/resources/message#embed-object-embed-field-structure). The contained objects should have `name` and `value` properties.    |
+| `replacements`   | N        | Map              | A map of strings to be replaced to objects containing `value` (and optionally `richValue`) that indicate what the template string should be replaced with for plain text and rich text. |
+
+We strongly recommend you include your plugin name within the `footer` so it is clear where notifications are coming from.
```

Comment:
I think I'd prefer if the `footer` is not possible to override & the `sourcePluginName` or w/e is a required field, so the requests _always_ include the plugin that sent it